### ISSUE TITLE
Strange depth map patterns

### ISSUE THREAD

**Issue by YessionCC** [ByteDance-Seed/Depth-Anything-3#125]
I ran the basic evaluation code. 
`
```python
import glob, os, torch
from depth_anything_3.api import DepthAnything3
device = torch.device("cuda")
model = DepthAnything3.from_pretrained("depth-anything/DA3NESTED-GIANT-LARGE")
model = model.to(device=device)
example_path = "assets/examples/SOH"
image_paths = sorted(glob.glob(os.path.join(example_path, "*.png")))
prediction = model.inference(
    image=image_paths,
    #process_res=1208,
    export_dir="./output",
    export_format="mini_npz-glb"
)
# prediction.processed_images : [N, H, W, 3] uint8   array
print(prediction.processed_images.shape)
# prediction.depth            : [N, H, W]    float32 array
print(prediction.depth.shape)  
# prediction.conf             : [N, H, W]    float32 array
print(prediction.conf.shape)  
# prediction.extrinsics       : [N, 3, 4]    float32 array # opencv w2c or colmap format
print(prediction.extrinsics.shape)
# prediction.intrinsics       : [N, 3, 3]    float32 array
print(prediction.intrinsics.shape)
```
`

Everything goes well but the visualization results are strange:

![Image](https://github.com/user-attachments/assets/3041ccc9-14f4-495f-b78f-163fbfef0c5e)

**Comment by changcongxun** [comment]
I have the same problem！Have you solved it？

**Comment by haotongl** [comment]
@changcongxun @YessionCC Could you please share any log or error messages from your run?  
Having those details would help me understand what went wrong and narrow down the issue.

**Comment by changcongxun** [comment]
> r

<img width="1563" height="322" alt="Image" src="https://github.com/user-attachments/assets/41cd005a-c984-41a8-92b7-c741b091653f" />

<img width="1337" height="728" alt="Image" src="https://github.com/user-attachments/assets/9c412280-c002-4817-bc89-87a091e1c616" />

**Comment by YessionCC** [comment]
There's no warning or error messages from my run
```markdown
[INFO ] using MLP layer as FFN
[INFO ] Processed Images Done taking 0.031548261642456055 seconds. Shape:  torch.Size([2, 3, 280, 504])
[INFO ] Model Forward Pass Done. Time: 63.22308421134949 seconds
[INFO ] Conversion to Prediction Done. Time: 0.0006999969482421875 seconds
[INFO ] conf_thresh_percentile: 40.0
[INFO ] num max points: 1000000
[INFO ] Exporting to GLB with num_max_points: 1000000
[INFO ] Export Results Done. Time: 0.1123194694519043 seconds
(2, 280, 504, 3)
(2, 280, 504)
(2, 280, 504)
(2, 3, 4)
(2, 3, 3)
```

**Comment by YessionCC** [comment]
And I found that each strange pattern is at 14*14 resolution. Maybe this issue is caused by image patch / interpolation /...

**Comment by Jaykumaran** [comment]
Hey,

I also found this issue, when i from using 
```


import torch
import numpy as np
import os
import sys
import cv2


 from depth_anything_3.api import DepthAnything3


def load_da3(encoder='da3metric-large', device='cuda'):
    print(f"[INFO] Initializing Depth Anything V3 model: {encoder} on {device}...")
    model = DepthAnything3(model_name=encoder)
    return model.to(device).eval()


@torch.no_grad()
def get_da3_depth(model, image_tensor):
    # Ensure H,W,3
    if image_tensor.shape[0] == 3:
        image_tensor = image_tensor.permute(1, 2, 0)

    img_np = (image_tensor.cpu().numpy() * 255).astype(np.uint8)
    h, w = img_np.shape[:2]

    # Inference
    prediction = model.inference(
        image=[img_np],
        process_res=1512,
        process_res_method="upper_bound_resize",
        export_dir=None,
        verbose=False
    )

    depth = prediction.depth[0]

    # Resize back
    if depth.shape != (h, w):
        depth = cv2.resize(depth, (w, h), interpolation=cv2.INTER_CUBIC)

    # Smooth + normalize
    d_min, d_max = depth.min(), depth.max()
    depth = (depth - d_min) / (d_max - d_min + 1e-6)

    return torch.from_numpy(depth).to(image_tensor.device)

```



It got resolved by using


```
def load_da3(encoder='da3metric-large', device='cuda'):
    print(f"[INFO] Initializing Depth Anything V3 model: {encoder} on {device}...")
    model = DepthAnything3.from_pretrained('depth-anything/DA3METRIC-LARGE')
    model = model.to(device).eval() 
    return model

@torch.no_grad()
def get_da3_depth(model, image_tensor):
    """
    Args:
        model: Loaded DA3 model
        image_tensor: (H, W, 3) Torch tensor, 0-1 range
    Returns:
        depth: (H, W) Torch tensor (Normalized 0-1)
    """
    # 1. Convert Tensor (0-1) to Numpy uint8 [0, 255]
    if image_tensor.shape[0] == 3: 
        image_tensor = image_tensor.permute(1, 2, 0)
    
    img_np = (image_tensor.cpu().numpy() * 255).astype(np.uint8)
    original_h, original_w = img_np.shape[:2]
    
    # 2. Run Inference
    # INCREASED RES: 1512 for High Quality. Lower to 1008 or 504 if OOM.
    prediction = model.inference(
        image=[img_np], 
        process_res = 1512,
        process_res_method = 'upper_bound_resize',
    ) # https://github.com/ByteDance-Seed/Depth-Anything-3/blob/81083bb002267a7d0368d2b687ac2c5c76a15468/src/depth_anything_3/app/modules/model_inference.py#L128C40-L128C58
    
    # 3. Get raw depth (Metric: Low=Close, High=Far)
    raw_depth = prediction.depth[0] 

    # 4. Resize to Original Resolution smoothly
    if raw_depth.shape[:2] != (original_h, original_w):
        depth = cv2.resize(raw_depth, (original_w, original_h), interpolation=cv2.INTER_CUBIC)

    depth = (depth - depth.min()) / (depth.max() - depth.min() + 1e-6)

    # 7. Convert to Tensor
    depth_tensor = torch.from_numpy(depth).to(image_tensor.device)
    
    return depth_tensor



From other script that I'm calling da3

                # Returns Metric Depth (Meters)
                metric_depth = get_da3_depth(model, gt_image)
                supervision_map = metric_depth


           if i_image == 0: 
                print("\n[DEBUG] Generating Depth Preview...")
                
                # 1. Convert Tensor to Numpy
                vis_img = supervision_map.squeeze().cpu().numpy()
                
                # 2. Re-Normalize to 0-255 for Visualization
                v_min, v_max = vis_img.min(), vis_img.max()
                print(f"[DEBUG] Depth Value Range: Min={v_min:.4f}, Max={v_max:.4f}")
                
                # if v_max - v_min > 1e-6:
                vis_img = (vis_img - v_min) / (v_max - v_min)
                # else:
                    # vis_img = np.zeros_like(vis_img)
                    
                vis_img = (vis_img * 255).astype(np.uint8)
                
                # 3. Apply ColorMap (Inferno is good for Depth)
                # vis_color = cv2.applyColorMap(vis_img, cv2.COLORMAP_INFERNO)
                
                # 4. Show Pop-up
                try:
                    cv2.namedWindow("Depth Preview (Press SPACE to continue)", cv2.WINDOW_NORMAL)
                    cv2.resizeWindow("Depth Preview (Press SPACE to continue)", 1024, 768)
                    cv2.imshow("Depth Preview (Press SPACE to continue)", vis_img)
                    print("[ACTION REQUIRED] Check the pop-up window. Press SPACE to continue training...")
                    cv2.waitKey(0) # Waits indefinitely
                    cv2.destroyAllWindows()
                except Exception as e:
                    # Fallback for headless servers (SSH without X11)
                    print(f"[WARN] Cannot open window (Error: {e}). Saving 'debug_preview.png' to current folder instead.")
                    cv2.imwrite("debug_preview.png", vis_img)


```

**Comment by YessionCC** [comment]
I resolve this issue by raplacing the "DA3NESTED-GIANT-LARGE" model with "DA3-LARGE" model. Now the depth results look more reasonable:

![Image](https://github.com/user-attachments/assets/895a2e6b-0b39-4d7f-87bd-caf09601b50e)
However, when I try the "DA3-LARGE" model, size mismatch error occurs and the model cannot be loaded correctly.

**Comment by Jaykumaran** [comment]
Hey,

> You can either preprocess all images to be of same size which you will be using as i/p 
> Or In my case directly passing og image as it is working without any error, 


The size mismatch is just a warning and not error, 

[WARN ] Images in batch have different sizes [(322, 504), (322, 504), (322, 504), (308, 504), (322, 504), (322, 504), (322, 504), (308, 504), (322, 504), (322, 504), (308, 504), (322, 504), (322, 504), (322, 504), (322, 504), (308, 504), (308, 504), (322, 504), (308, 504), (308, 504), (322, 504), (308, 504), (322, 504), (308, 504), (308, 504), (308, 504), (308, 504), (322, 504), (308, 504), (322, 504), (322, 504)]; **center-cropping all to smallest (308,504)**

```

import glob, os, torch
from depth_anything_3.api import DepthAnything3
device = torch.device("cuda")
model = DepthAnything3.from_pretrained("depth-anything/DA3NESTED-GIANT-LARGE")
model = model.to(device=device)
example_path = "/chunk_6/images"
images = sorted(glob.glob(os.path.join(example_path, "*.jpg")))
prediction = model.inference(
    images,
    process_res = 504, # 1008, 1512
    process_res_method = 'upper_bound_resize'
)


# prediction.processed_images : [N, H, W, 3] uint8   array
print(prediction.processed_images.shape)
# prediction.depth            : [N, H, W]    float32 array
print(prediction.depth.shape)  
# # prediction.conf             : [N, H, W]    float32 array
# print(prediction.conf.shape)  
# # prediction.extrinsics       : [N, 3, 4]    float32 array # opencv w2c or colmap format
# print(prediction.extrinsics.shape)
# # prediction.intrinsics       : [N, 3, 3]    float32 array
# print(prediction.intrinsics.shape)


import cv2
import numpy as np
import os

# Directory where depth images will be saved (same as input folder)
save_dir = example_path

depths = prediction.depth  # [N, H, W] float32
conf = prediction.conf
processed_imgs = prediction.processed_images

for img_path, depth in zip(images, depths):
    # Normalize depth to 0–65535 for 16-bit PNG
    depth_norm = depth - depth.min()
    depth_norm = depth_norm / depth_norm.max()
    depth_16 = (depth_norm * 65535.0).astype(np.uint16)

    # Output filename: originalName_depth.png
    base = os.path.basename(img_path)
    name, _ = os.path.splitext(base)
    out_path = os.path.join(save_dir, f"{name}_depth.png")

    # Save 16-bit depth PNG
    cv2.imwrite(out_path, depth_16)

    print("Saved:", out_path)


```



As I was mentioning using  DepthAnything3(model_name=encoder) has some issues whereas 

`DepthAnything3.from_pretrained("depth-anything/DA3NESTED-GIANT-LARGE")` is working without weird patchy depth. Kindly run this once and check, Hope this helps.

**Comment by YessionCC** [comment]
My codes are working well now after replacing `self.model_name=model name` by `self.model_name = kwargs['config']['model_name']` in line 85 of `src/depth_anything_3/api.py`. I found the `model_name` is always `da3-large` even if I use other models.

<img width="1179" height="585" alt="Image" src="https://github.com/user-attachments/assets/f0562641-2670-4bb7-b727-cec40d0c63a0" />

**Comment by changcongxun** [comment]
> I resolve this issue by raplacing the "DA3NESTED-GIANT-LARGE" model with "DA3-LARGE" model. Now the depth results look more reasonable:
> 
> ![Image](https://github.com/user-attachments/assets/895a2e6b-0b39-4d7f-87bd-caf09601b50e) However, when I try the "DA3-LARGE" model, size mismatch error occurs and the model cannot be loaded correctly.

"Great job, but the depth of the sky at the top of the image seems incorrect. Any suggestions for improvement?"

**Comment by YessionCC** [comment]
Using "DA3NESTED-GIANT-LARGE" model can resolve the issue. It segments the sky and replaces the depth of sky with a very large value.

**Comment by isxiaohe** [comment]
> Using "DA3NESTED-GIANT-LARGE" model can resolve the issue. It segments the sky and replaces the depth of sky with a very large value.

Hi, I'm using "DA3NESTED-GIANT-LARGE" model but I get no sky segmentation output. Do you have the same problem? Thank you for your respond!
